### PR TITLE
Use new braced route element placeholders in resource paths.

### PR DESCRIPTION
**Pull request by ADmad**

### CODE REVIEW

File: src/Routing/RouteBuilder.php [hunk 1]
```diff
@@ -215,6 +215,11 @@ public function addExtensions($extensions)
      */
     public function path(): string
     {
+        $routeKey = strpos($this->_path, '{');
+        if ($routeKey !== false && strpos($this->_path, '}') !== false) {
```
Comment (othercorey) [inline src/Routing/RouteBuilder.php:219]: Shouldn't this look for `}` after `{`?

Comment (ADmad) [reply src/Routing/RouteBuilder.php:219]: That's what's it's doing.

Comment (othercorey) [reply src/Routing/RouteBuilder.php:219]: I mean after the position in the string.

Comment (ADmad) [reply src/Routing/RouteBuilder.php:219]: Ah, ideally yes. I copied the check from another place in the routing code. Guess I can update all, though if one has a closing brace before an opening brace then their route template is most likely messed up.

Comment (othercorey) [reply src/Routing/RouteBuilder.php:219]: Up to you if you want to throw a validation error if } is before {.

Comment (ADmad) [reply src/Routing/RouteBuilder.php:219]: I don't think such a check is necessary. A broken route would be indication enough. Being over zealous in handling dev error didn't go too well last time we tried :wink:.

File: src/Routing/RouteBuilder.php [hunk 2]
```diff
@@ -421,7 +426,7 @@ public function resources(string $name, $options = [], $callback = null)
 
         if ($callback !== null) {
             $idName = Inflector::singularize(Inflector::underscore($name)) . '_id';
-            $path = '/' . $options['path'] . '/:' . $idName;
+            $path = '/' . $options['path'] . '/{' . $idName . '}';
```
Comment (garas) [inline src/Routing/RouteBuilder.php:429]: 4 concatenations is bit too much, better `sprintf`

Comment (ADmad) [reply src/Routing/RouteBuilder.php:429]: The total line length is quite reasonable and it's still easy enough to grok what's being done :).

**Pull request merged**